plugin.py

```diff
@@ -62,15 +59,19 @@ def __init__(self, session: "weakref.ref[Session]") -> None:
         if sess:
             self.plugin_mapping[sess.window.id()] = self
 
-        # ST persists view setting after getting closed so we have to reset some status
+        # ST persists view settings after getting closed so we have to reset some status
         for window in sublime.windows():
             for view in window.views(include_transient=True):
-                erase_copilot_view_setting(view, "is_visible")
-                erase_copilot_view_setting(view, "is_waiting")
+                cm = ViewCompletionManager(view)
+                cm.is_visible = False
+                cm.is_waiting = False
+
+                pcm = ViewPanelCompletionManager(view)
+                pcm.is_waiting = False
 
     def on_ready(self, api: ApiWrapperInterface) -> None:
         def on_check_status(result: CopilotPayloadSignInConfirm, failed: bool) -> None:
-            self.set_has_signed_in(result.get("status") == "OK")
+            self.set_has_signed_in(result["status"] == "OK")
```

Comment:
There is no guarantee that the message will have a status here and we don't handle it properly with this approach. One example is if there is a `self-signed` certificate in the chain it will return the following:

```json
{'message': 'Request checkStatus failed with message: self signed certificate in certificate chain', 'code': -32603}
```

Reply:
> There is no guarantee that the message will have a status here and we don't handle it properly with this approach. One example is if there is a `self-signed` certificate in the chain it will return the following:
> 
> ```json
> {
> 	"message": "Request checkStatus failed with message: self signed certificate in certificate chain", 	 
>         "code": -32603
> }
> ```

But I think then they can't use this plugin?

Reply:
Won't work